### PR TITLE
Simplify and improve consensus payload serialization and deserialization

### PR DESCRIPTION
**Pull request by vncoelho**
Reducing two `writer.WriteVarInt` and creating a two functions for writer and reader in order to avoid repeating the code.

### CODE REVIEW

File: neo/Consensus/ConsensusContext.cs [hunk 1]
```diff
@@ -417,37 +412,27 @@ public void Serialize(BinaryWriter writer)
...
+        {
+            foreach (var payload in payloadArray)
             {
                 bool hasPayload = !(payload is null);
                 writer.Write(hasPayload);
                 if (!hasPayload) continue;
                 writer.Write(payload);
             }
         }
+        private void ReadConsensusPayloadArray(BinaryReader reader, out ConsensusPayload[] payloadArray, ulong payloadLenght)
```
Comment (erikzhang) [inline neo/Consensus/ConsensusContext.cs]: Why use output parameter instead of return value?

Comment (vncoelho) [reply neo/Consensus/ConsensusContext.cs]: Let's change

Comment (vncoelho) [reply neo/Consensus/ConsensusContext.cs]: Done.

**Comment by shargon** [comment]
Wait, I will add one UT and approve it.

**Comment by vncoelho** [comment]
Such a great job! Looks better

**Pull request merged**